virttest/utils_libvirt/libvirt_pcicontr.py

```diff
@@ -28,3 +28,27 @@ def get_max_contr_indexes(vm_xml, cntlr_type, cntlr_model, cntl_num=1):
                                                          cntlr_model,
                                                          usable_indexes[:cntl_num]))
     return usable_indexes[:cntl_num]
+
+
+def get_free_pci_slot(vm_xml, target_bus, max_slot=30):
```

Comment:
Do you need controller  type and model as filter conditions?

Reply:
@chunfuwen currently, it seems this function only applies to the controller 'pcie-root' because pcie-root-port which is common only can accept 1 pci device. Only pcie-root can has 31 slots available.  I will update the docstring to make it clear.